crates/re_renderer/src/wgpu_resources/dynamic_resource_pool.rs

```diff
@@ -70,18 +96,33 @@ where
             // Otherwise create a new resource
             } else {
                 let resource = creation_func(desc);
-                self.total_resource_size_in_bytes += desc.resource_size_in_bytes();
-                Arc::new(self.resources.insert((desc.clone(), resource)))
+                self.total_resource_size_in_bytes.fetch_add(
+                    desc.resource_size_in_bytes(),
+                    std::sync::atomic::Ordering::Relaxed,
+                );
+                resource
             };
 
-        self.alive_handles.insert(*handle, Arc::clone(&handle));
-        handle
+        let handle = state.alive_resources.insert_with_key(|handle| {
+            Some(Arc::new(DynamicResource {
+                inner: inner_resource,
+                creation_desc: desc.clone(),
+                handle,
+            }))
+        });
+
+        state.alive_resources[handle].as_ref().unwrap().clone()
```

Comment:
this can be re-written without the `unwrap` by creating the resource and storing a clone in the `alive_resources`

Reply:
Struggling with that since the resource itself contains the handle.
There might be a follow-up PR where we no longer use slotmaps for this which will eliminate this problem

Reply:
oh right, I missed that